vault/identity_store_oidc.go

```diff
@@ -924,6 +943,45 @@ func (tok *idToken) generatePayload(logger hclog.Logger, template string, entity
 	return payload, nil
 }
 
+func populateTemplates(claims map[string]interface{}, logger hclog.Logger, ns *namespace.Namespace, entity *identity.Entity, groups []*identity.Group, templates ...string) error {
+	for _, template := range templates {
+		// Parse and integrate the populated template. Structural errors with the template _should_
+		// be caught during configuration. Error found during runtime will be logged, but they will
+		// not block generation of the basic ID token. They should not be returned to the requester.
+		_, populatedTemplate, err := identitytpl.PopulateString(identitytpl.PopulateStringInput{
+			Mode:        identitytpl.JSONTemplating,
+			String:      template,
+			Entity:      identity.ToSDKEntity(entity),
+			Groups:      identity.ToSDKGroups(groups),
+			NamespaceID: ns.ID,
+		})
+		if err != nil {
+			logger.Warn("error populating OIDC token template", "template", template, "error", err)
+		}
+
+		if populatedTemplate != "" {
+			var parsed map[string]interface{}
+			if err := json.Unmarshal([]byte(populatedTemplate), &parsed); err != nil {
+				logger.Warn("error parsing OIDC template", "template", template, "err", err)
+			}
+
+			for k, v := range parsed {
+				if _, conflict := claims[k]; conflict {
+					return fmt.Errorf("found conflicting top level OIDC claim key %q", k)
```

Comment:
Is the comment above accurate since we can return an error? Would it be useful to also indicate which are the two templates that contain the same claims?